fabric-message-api-v1/src/client/java/net/fabricmc/fabric/mixin/client/message/MessageHandlerMixin.java

```diff
@@ -0,0 +1,82 @@
+	private void fabric_onSignedChatMessage(MessageType.Parameters params, SignedMessage message, Text decorated, GameProfile sender, boolean onlyShowSecureChat, Instant receptionTimestamp, CallbackInfoReturnable<Boolean> cir) {
+		fabric_onChatMessage(decorated, message, sender, params, receptionTimestamp, cir);
+	}
+
+	@Inject(method = "processChatMessageInternal", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/ChatHud;addMessage(Lnet/minecraft/text/Text;Lnet/minecraft/network/message/MessageSignatureData;Lnet/minecraft/client/gui/hud/MessageIndicator;)V", ordinal = 1), cancellable = true)
+	private void fabric_onFilterSignedChatMessage(MessageType.Parameters params, SignedMessage message, Text decorated, GameProfile sender, boolean onlyShowSecureChat, Instant receptionTimestamp, CallbackInfoReturnable<Boolean> cir) {
+		Text filtered = message.filterMask().getFilteredText(message.getSignedContent());
+
+		if (filtered != null) {
+			fabric_onChatMessage(params.applyChatDecoration(filtered), message, sender, params, receptionTimestamp, cir);
+		}
+	}
+
+	@Inject(method = "method_45745", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/ChatHud;addMessage(Lnet/minecraft/text/Text;)V"), cancellable = true)
+	private void fabric_onProfilelessChatMessage(MessageType.Parameters params, Text decorated, Instant receptionTimestamp, CallbackInfoReturnable<Boolean> cir) {
+		fabric_onChatMessage(decorated, null, null, params, receptionTimestamp, cir);
+	}
+
+	@Unique
+	private void fabric_onChatMessage(Text message, @Nullable SignedMessage signedMessage, @Nullable GameProfile sender, MessageType.Parameters params, Instant receptionTimestamp, CallbackInfoReturnable<Boolean> cir) {
+		if (ClientReceiveMessageEvents.ALLOW_CHAT.invoker().allowReceiveChatMessage(message, signedMessage, sender, params, receptionTimestamp)) {
+			ClientReceiveMessageEvents.CHAT.invoker().onReceiveChatMessage(message, signedMessage, sender, params, receptionTimestamp);
+		} else {
+			ClientReceiveMessageEvents.CHAT_CANCELED.invoker().onReceiveChatMessageCanceled(message, signedMessage, sender, params, receptionTimestamp);
+			cir.setReturnValue(false);
+		}
+	}
+
+	@Inject(method = "onGameMessage", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/InGameHud;setOverlayMessage(Lnet/minecraft/text/Text;Z)V"), cancellable = true)
+	private void fabric_onOverlayGameMessage(Text message, boolean overlay, CallbackInfo ci) {
+		fabric_onGameMessage(message, overlay, ci);
+	}
+
+	@Inject(method = "onGameMessage", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/ChatHud;addMessage(Lnet/minecraft/text/Text;)V"), cancellable = true)
+	private void fabric_onGameMessage(Text message, boolean overlay, CallbackInfo ci) {
```

Comment:
You should be able to use a `JUMP` injection point to inject before the `if (overlay)`. Not sure if it's worth it though. Otherwise you can specify two selectors in the `at` to only have a single method.